src/DGtal/io/colormaps/TickedColorMap.h

```diff
@@ -0,0 +1,242 @@
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+  
+  /////////////////////////////////////////////////////////////////////////////
+  // template class TickedColorMap
+  /**
+   * Description of template class 'TickedColorMap' <p>
+   *
+   * @brief Aim: This class adapts any colormap to add "ticks" in the
+   * colormap colors.
+   *
+   * Ticks are characterized by a position and a width in the adapted
+   * colormap range. When the value falls in a tick, the colormap color is
+   * replaced by the color provided in the constructor.
+   *
+   * For example, on a colormap, this class can be used to create white
+   * ticks with regular spacing, or locate zero-crossing of a curvature map
+   * for instance (i.e. with a single tick at zero).
+   *
+   * @tparam TValue The type of the range values.
+   * @tparam TColorMap The type of colormap to Adapt (the value type of TColorMap must be TValue).
+   *
+   */
+  template <typename TValue, typename TColorMap>
+  class TickedColorMap
+  {
+    
+  public:
+    
+    ///Value type
+    typedef TValue Value;
+    
+    ///Adapted colormap type.
+    typedef TColorMap ColorMap;
+    
+    BOOST_CONCEPT_ASSERT(( concepts::CColorMap<ColorMap> )) ;
+    BOOST_STATIC_ASSERT(( ConceptUtils::SameType<Value, typename ColorMap::Value>::value));
+    
+    // ----------------------- Standard services ------------------------------
+  public:
+    
+    /**
+     * Constructor.
+     *
+     * @pre aMin < aMax.
+     *
+     * @param aMin The lower bound of the value range.
+     * @param aMax The upper bound of the value range.
+     * @param color the color of the ticks.
+     */
+    TickedColorMap(const Value & aMin,
+                   const Value & aMax,
+                   const Color &color = Color::White);
+    
+    /**
+     * Computes the color associated with a value in a given range.
+     *
+     * @note This operation has a complexity in @f$ log(N)@f$ where
+     * @f$ N@f$ is the number of ticks.
+     *
+     * @param value A value within the value range.
+     * @return A color.
+     */
+    Color operator()( const Value & value ) const;
+    
+    /**
+     * Destructor.
+     */
+    ~TickedColorMap();
+    
+    /**
+     * Copy constructor.
+     * @param other the object to clone.
+     */
+    TickedColorMap ( const TickedColorMap & other );
+    
+    /**
+     * Assignment.
+     * @param other the object to copy.
+     * @return a reference on 'this'.
+     */
+    TickedColorMap & operator= ( const TickedColorMap & other );
+    
+    // ----------------------- Interface --------------------------------------
+  public:
+    
+    /**
+     * Add a tick at a given position of the
+     * range.
+     *
+     * @param position tick position.
+     * @param thickness tick thickness.
+     */
+    void addTick(const Value position,
+                 const Value thickness);
+    
+    /**
+     * Add regular ticks in the range.
```

Comment:
Yes, you can do as many addticks and andRegularticks as you want. The finalize() will make everything consistent